src/components/BookmarkManager.vue

```diff
@@ -0,0 +1,180 @@
+<script setup lang="ts">
+import { ref, computed, onMounted } from 'vue'
+
+interface Bookmark {
+  url: string
+  name: string
+  isRenaming: boolean
+}
+
+const bookmarks = ref<Bookmark[]>([])
+
+onMounted(() => {
+  requestBookmark()
+})
+
+const requestBookmark = () => {
+  const response = window.prompt('VERSO::BOOKMARK_LIST_GET')
+  if (response !== null) {
+    const status = JSON.parse(response) as { [url: string]: Bookmark }
+    bookmarks.value = Object.values(status)
+  }
+}
+
+const onclickRemove = (id: string) => {
+  window.prompt('VERSO::BOOKMARK_REMOVE::' + id)
+}
+
+const EndRename = (id: string, name: string) => {
+  const bookmark = bookmarks.value.find(b => b.url === id)
+  if (bookmark) {
+    bookmark.isRenaming = false
+    window.prompt('VERSO::BOOKMARK_RENAME::' + id + '::' + name)
+  }
+}
+const onclickStartRename = (id: string) => {
+  if (bookmarks.value.some(b => b.isRenaming)) {
+    return
+  }
+  const bookmark = bookmarks.value.find(b => b.url === id)
+  if (bookmark) {
+    bookmark.isRenaming = true
```

Comment:
Should also disable or hide other rename, remove buttons since you're not allowing actions when renaming.